src/components/list-blog/ListBlogComponent.tsx

```diff
@@ -0,0 +1,33 @@
+import axios from 'axios';
+import React, {useState, useEffect} from 'react' 
+import PaginationComponent from '../pagination/PaginationComponent';
+import PostComponent from '../post/PostsComponent';
+
+
+export default function ListBlogComponent() {
+    const [posts, setPosts] = useState([]);
+    const [currentPage, setCurrentpage] = useState(1);
+    const [postsPerPage, setPostsPerPage] = useState(10);
+
+
+    useEffect(() => {
+        const fetchPosts = async () => {
+            const res = await axios.get('https://jsonplaceholder.typicode.com/posts');
+            setPosts(res.data);
+        }
+        fetchPosts();
+    },[]);
+   
+    const indexOfLastPost = currentPage * postsPerPage;
```

Comment:
You should make a folder called `src/hooks/blogHooks.ts`, create a function called `useListBlog`, move all states and calculate define like this to that function, and return only what you need. Ex: `currentPosts`, `postsPerPage `, `postsLenght`, `paginate`